### PR TITLE
feat(accesscontextmanager): generate library

### LINKED ISSUE #7851: Access Context Manager API: `accesscontextmanager.googleapis.com`

### PR DESCRIPTION
**Pull request by coryan**
Fixes #7851

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7856)

### CODE REVIEW

File: google/cloud/accesscontextmanager/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+* Bazel and CMake builds
+
+## Documentation
+
+* Official documentation about the [Access Context Manager API][cloud-service-docs] service
+* [Reference doxygen documentation][doxygen-link] for each release of this
+  client library
+* Detailed header comments in our [public `.h`][source-link] files
+
+[cloud-service-docs]: https://cloud.google.com/accesscontextmanager
```
Comment (dbolduc) [inline google/cloud/accesscontextmanager/README.md:29]: 404. You want: https://cloud.google.com/access-context-manager/docs

File: google/cloud/accesscontextmanager/doc/main.dox [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+For operations that do not return a value the library simply returns
+`google::cloud::Status`.
+
+### Retry, Backoff, and Idempotency Policies.
+
+The library automatically retries requests that fail with transient errors, and
+uses [exponential backoff] to backoff between retries. Application developers
+can override the default policies.
+
+[cloud-service-docs]: https://cloud.google.com/accesscontextmanager
```
Comment (dbolduc) [inline google/cloud/accesscontextmanager/doc/main.dox:87]: ditto

File: google/cloud/accesscontextmanager/doc/main.dox [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+[exponential backoff]: https://en.wikipedia.org/wiki/Exponential_backoff
+[gcs-quickstart]: https://cloud.google.com/accesscontextmanager/docs/quickstarts 'Quickstarts'
```
Comment (dbolduc) [inline google/cloud/accesscontextmanager/doc/main.dox:89]: link is broken, but we actually don't use `gcs-quickstart` anywhere.... we should remove from the scaffolding.

File: google/cloud/accesscontextmanager/quickstart/README.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+```console
+@powershell -NoProfile -ExecutionPolicy unrestricted -Command ^
+    (new-object System.Net.WebClient).Downloadfile( ^
+        'https://pki.google.com/roots.pem', 'roots.pem')
+set GRPC_DEFAULT_SSL_ROOTS_FILE_PATH=%cd%\roots.pem
+```
+
+[bazel-install]: https://docs.bazel.build/versions/main/install.html
+[quickstart-link]: https://cloud.google.com/accesscontextmanager/docs/quickstart
```
Comment (dbolduc) [inline google/cloud/accesscontextmanager/quickstart/README.md:160]: 404. You want: https://cloud.google.com/access-context-manager/docs/quickstart

**Review by dbolduc — CHANGES_REQUESTED** [review]

File: ci/cloudbuild/builds/cmake-install.sh [hunk 5]
```diff
@@ -96,6 +96,9 @@ expected_dirs+=(
   ./include/google/devtools/cloudtrace
   ./include/google/devtools/cloudtrace/v2
   ./include/google/iam/v1
+  # This contains pure protos. The generator does not
+  # updated expected_install_directories in this case.
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/cloudbuild/builds/cmake-install.sh]: Done.

File: google/cloud/accesscontextmanager/README.md (hunk 1 above)
Comment (coryan) [reply google/cloud/accesscontextmanager/README.md]: Fixed here and below.

File: google/cloud/accesscontextmanager/doc/main.dox (hunk 3 above)
Comment (coryan) [reply google/cloud/accesscontextmanager/doc/main.dox]: Removed. I am sending a separate PR to fix the generator.

File: google/cloud/accesscontextmanager/quickstart/README.md (hunk 4 above)
Comment (coryan) [reply google/cloud/accesscontextmanager/quickstart/README.md]: Fixed.

File: google/cloud/accesscontextmanager/doc/main.dox (hunk 2 above)
Comment (coryan) [reply google/cloud/accesscontextmanager/doc/main.dox]: Fixed.

File: ci/cloudbuild/builds/cmake-install.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (dbolduc) [reply ci/cloudbuild/builds/cmake-install.sh]: Thanks

**Review by dbolduc — APPROVED** [review]

**Pull request merged**